File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/Query_Nodes_Arguments.java
package com.example.grts;

import graphql.schema.GraphQLInputObjectType;
import viaduct.java.api.context.ExecutionContext;
import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;
import viaduct.apiannotations.InternalApi;
import viaduct.java.api.types.Arguments;
import viaduct.java.api.internal.InputBase;
import viaduct.java.api.internal.InternalContext;

/** Generated arguments class for resolver field. */
public class Query_Nodes_Arguments extends InputBase implements Arguments {

    public static final Type<Query_Nodes_Arguments> Reflection = Type.ofClass(Query_Nodes_Arguments.class);

    public static final class Fields implements TypeFields<Query_Nodes_Arguments> {
        private Fields() {}

        public static final Field<Query_Nodes_Arguments> __typename =
                Field.of("__typename", Reflection);
                public static final Field<Query_Nodes_Arguments> ids =
                                Field.of("ids", Reflection);

    }

    // Public because the framework constructs arguments reflectively across packages
    // (JavaFieldResolverExecutorImpl, VariablesProviderExecutorImpl, etc.). @InternalApi
    // marks it as not-for-tenant-use, mirroring Kotlin's `internal constructor`.
    @InternalApi
    public Query_Nodes_Arguments(InternalContext context, Map<String, Object> data, GraphQLInputObjectType graphQLInputObjectType) {
        super(context, data, graphQLInputObjectType);
    }

    /**
     * Returns whether this input contains a value for {@code field} after GraphQL
     * defaults are applied. Explicit {@code null} counts as present. An omitted field
     * with a schema default is present; an omitted field without a default is absent.
     */
    public boolean isPresent(Field<Query_Nodes_Arguments> field) {
        return isFieldPresent(field);
    }

        public List<String> getIds() {
            return getScalarList("ids");
        }


    public static Builder builder(ExecutionContext context) {
        return new Builder(InternalContext.from(context));
    }

    public static class Builder {
        private final InternalContext __context;
        private final GraphQLInputObjectType graphQLInputObjectType;
        private final Map<String, Object> data = new LinkedHashMap<>();

        private Builder(InternalContext __context) {
            this.__context = __context;
            this.graphQLInputObjectType =
                    __context.getArgumentsInputType(
                            "Query_Nodes_Arguments",
                            "Query",
                            "nodes");
        }

                public Builder ids(List<String> ids) {
                    data.put("ids", ids);
        return this;
                }


        public Query_Nodes_Arguments build() {
            return new Query_Nodes_Arguments(
                    __context, new LinkedHashMap<>(data), graphQLInputObjectType);
        }
    }
}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/Query_Nodes_Arguments.kt
@file:Suppress("warnings")

package viaduct.api.grts

import graphql.schema.GraphQLInputObjectType
import viaduct.apiannotations.InternalApi
import viaduct.api.context.ExecutionContext
import viaduct.api.internal.InputTypeFactory
import viaduct.api.internal.InputValueBuilder
import viaduct.api.internal.InternalContext
import viaduct.api.internal.internal
import viaduct.api.internal.InputLikeBase
import viaduct.api.types.Input

@OptIn(InternalApi::class)
class Query_Nodes_Arguments internal constructor(
    override val context: InternalContext,
    override val inputData: Map<String, Any?>,
    override val graphQLInputObjectType: GraphQLInputObjectType,
): InputLikeBase(), viaduct.api.types.Arguments {
    init {
       TODO()
    }

    val ids: kotlin.collections.List<kotlin.String> get() = TODO()


    fun toBuilder() = Builder(context, graphQLInputObjectType, this.inputData.toMutableMap())

    object of {
        operator fun invoke(context: ExecutionContext, block: Builder.() -> Unit): Query_Nodes_Arguments =
            Builder(context).apply(block).build()
    }

    class Builder internal constructor(
        override val context: InternalContext,
        override val graphQLInputObjectType: GraphQLInputObjectType,
        override val inputData: MutableMap<String, Any?> = TODO()
    ) : InputLikeBase.Builder(), InputValueBuilder<Query_Nodes_Arguments> {

        constructor(context: ExecutionContext): this(
            context.internal,
            InputTypeFactory.argumentsInputType("Query_Nodes_Arguments", "Query", "nodes", context.internal.schema),
            mutableMapOf()
        )

        init {
            TODO()
        }

                    fun ids(value: kotlin.collections.List<kotlin.String>): Builder = TODO()


        final override fun build(): Query_Nodes_Arguments = TODO()
    }

    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.Query_Nodes_Arguments> {
        override final val name = "Query_Nodes_Arguments"
        override final val kcls = viaduct.api.grts.Query_Nodes_Arguments::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.Query_Nodes_Arguments> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.Query_Nodes_Arguments> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.Query_Nodes_Arguments.Reflection)

            final val ids: viaduct.api.reflect.Field<viaduct.api.grts.Query_Nodes_Arguments> =
                viaduct.api.internal.FieldImpl("ids", viaduct.api.grts.Query_Nodes_Arguments.Reflection)

    }

}